packages/notebook/src/widget.ts

```diff
@@ -1207,13 +1207,21 @@ export class Notebook extends StaticNotebook {
   /**
    * Construct a notebook widget.
    */
+
   constructor(options: Notebook.IOptions) {
     super(Private.processNotebookOptions(options));
     this.node.tabIndex = 0; // Allow the widget to take focus.
     // Allow the node to scroll while dragging items.
     this.node.setAttribute('data-lm-dragscroll', 'true');
   }
 
+  get selectedCells(): Cell[] {
+    let cellList: Cell[] = this.widgets.filter(cell =>
+      this.isSelectedOrActive(cell)
+    );
+    return cellList;
```

Comment:
Could you set up a cache to avoid looping on the cells for each get?

You need to add a private variable `_cellList` and update that list when the signals `selectionChanged` and `activeCellChanged`.

```typescript
constructor(options: Notebook.IOptions) {
  // ... existing code
}

/**
 * The active and selected cells
 */
get selectedCells(): Cell[] {
  return this._selectedCells
}

private _updateSelectedCells(): void {
  this._selectedCells = this.widgets.filter(cell =>
      this.isSelectedOrActive(cell)
    );
}

private _selectedCells: Cell[] = [];
```

Reply:
Thanks for your comments @fcollonval They are included in [84efb7a](https://github.com/jupyterlab/jupyterlab/pull/13861/commits/84efb7acf463b1310b538c1e6397768b18ea5184).